src/store/outbox/actions.js

```diff
@@ -105,4 +108,33 @@ export default {
 
 		commit('deleteMessage', { id })
 	},
+
+	async sendMessageWithUndo({ getters, dispatch }, { message }) {
```

Comment:
nit: the action is async but can't be awaited. You could return a new promise at the end that calls the `resolve` callback when everything is done